pkg/api/folder.go

```diff
@@ -84,8 +86,22 @@ func UpdateFolder(c *models.ReqContext, cmd models.UpdateFolderCommand) response
 	return response.JSON(200, toFolderDto(g, cmd.Result))
 }
 
-func DeleteFolder(c *models.ReqContext) response.Response {
+func (hs *HTTPServer) DeleteFolder(c *models.ReqContext) response.Response { // temporary adding this function to HTTPServer, will be removed from HTTPServer when librarypanels featuretoggle is removed
 	s := dashboards.NewFolderService(c.OrgId, c.SignedInUser)
+	if hs.Cfg.IsPanelLibraryEnabled() {
+		f, err := s.GetFolderByUID(c.Params(":uid"))
+		if err != nil {
+			return toFolderError(err)
+		}
+		err = hs.LibraryPanelService.DeleteLibraryPanelsInFolder(c, f)
+		if errors.Is(err, librarypanels.ErrFolderHasConnectedLibraryPanels) {
+			return response.Error(500, "Folder could not be deleted because it contains linked library panels", err)
+		}
+		if err != nil {
+			hs.log.Error("Failed to delete library panels in folder", "folder", f.Id, "user", c.SignedInUser.UserId, "error", err)
+		}
```

Comment:
I think the following pattern, of first checking if `err` is non-nil, and then handling error variants, is much more readable (because you don't have to read several blocks to ensure all errors are handled):
```suggestion
		if err != nil {
			if errors.Is(err, librarypanels.ErrFolderHasConnectedLibraryPanels) {
				return response.Error(500, "Folder could not be deleted because it contains linked library panels", err)
			}
			
			hs.log.Error("Failed to delete library panels in folder", "folder", f.Id, "user", c.SignedInUser.UserId, "error", err)
		}
```

Reply:
Excellent suggetion, much more readable!